index.js

```diff
@@ -25,8 +25,14 @@ function _fillMemDuplex(memDuplexes, index, dataRetrievalFn, response, logger){
     return dataRetrievalFn(memDuplexes[index].location, logger,
         (err, readable) => {
         if(err){
-            logger.error('failed to get full object');
-            return response.end();
+            logger.error('failed to get full object', {
+                error: err,
+                method: '_fillMemDuplex',
+            });
+            if (response.connection) {
+                return response.connection.destroy();
+            }
+            throw new Error('failed to get full object');
```

Comment:
Do we still have to throw? We are already destroying the socket, that should be enough for the errored request.